orangecontrib/text/preprocess.py

```diff
@@ -1,94 +1,298 @@
+            self.use_df_sw = True
+            is_float = isinstance(max_df, float)
+            is_int = isinstance(max_df, int)
+            if is_float:
+                in_range = 1.0 >= max_df >= 0.0
+                if not in_range:
+                    raise ValueError('Parameter max_df is out of range ({}).'
+                                     .format(max_df))
+            elif not is_int:
+                raise ValueError('Type "{}" not supported.'.format(max_df))
 
 
 class Stemmatizer():
-    """
-        A common class for stemming and lemmatization.
-    """
-    def __init__(self, trans, lowercase=True, name='Stemmatizer'):
+    """A common class for stemming and lemmatization."""
+    def __init__(self, transformation, name='Stemmatizer'):
         """
-            :param trans: The method that will perform transformation on the tokens.
-            :param name: The name of the transformation object.
-            :return: :class: `orangecontrib.text.preprocess.Stemmatizer`
+        Args:
+            transformation (function): The function that will perform
+                transformation on the tokens.
+            name (Optional[str]): The name of the transformation object.
+
+        Returns:
+            `orangecontrib.text.preprocess.Stemmatizer`: The instance of a
+                Stemmatizer of choice.
         """
-        self.trans = trans
+        self.transformation = transformation
         self.name = name
-        self.lowercase = lowercase
 
     def __call__(self, data):
-        """
-            :param data: The input that we wish to transform.
-            :type data: string, list or :class: `Orange.data.Table`
-            :return: string, list or :class: `Orange.data.Table`
+        """Calls the transformation function of this instance.
+
+        Args:
+            data (str, list): The input that we wish to transform.
+
+        Returns:
+            str, list: The transformed input in the form of a string or list,
+                depending on the input.
+
+        Raises:
+            ValueError: If the input type is not supported.
         """
         if isinstance(data, str):
-            if self.lowercase:
-                return self.trans(data).lower()
-            else:
-                return self.trans(data)
+            return self.transformation(str)
         elif isinstance(data, list):
-            if self.lowercase:
-                return [self.trans(word).lower() for word in data]
-            else:
-                return [self.trans(word) for word in data]
+            return [self.transformation(word) for word in data]
         else:
-            raise ValueError("Type {} not supported.".format(type(data)))
+            raise ValueError('Type {} not supported.'.format(type(data)))
 
-Stemmer = Stemmatizer(PorterStemmer().stem, 'Stemmer')
+PorterStemmer = Stemmatizer(PorterStemmer().stem, 'PorterStemmer')
+SnowballStemmer = Stemmatizer(SnowballStemmer(language="english").stem,
```

Comment:
@nikicc I've been thinking about the language option and I'd appreciate some further input on the matter. I think we would best benefit from adding a new parameter to the Preprocess class, that determines the language. Language is then subsequently used in stop word removal and transformations. In that aspect we would have to rework the stop word parameter from (str, list or None) to something else, possibly two new parameters (use_stopwords=True/False, custom_stopwords=[...]) and additionaly the aforementioned language parameter. Stemmatizers would then also be generated in a Preprocess class getter method, since we need the language to create them (but I will check on availability for other languages as well). Would this reformatting make sense?